server/bin/pbench-server-prep-shim-002.py

```diff
@@ -110,147 +79,41 @@ def process_tb(config, logger, receive_dir, qdir_md5, duplicates, errors):
         os.path.join(receive_dir, "**", "*.tar.xz.md5"), recursive=True
     )
 
-    archive = config.ARCHIVE
     logger.info("{}", config.TS)
     list_check.sort()
     nstatus = ""
 
-    ntotal = ntbs = nerrs = nquarantined = ndups = 0
+    ntotal = ntbs = nerrs = 0
+
+    # FIXME: keep a generated token which has longer duration
+
+    token = config.get("put-server", "token")
+    move_results = MoveResults(config, token, logger)
 
     for tbmd5 in list_check:
         ntotal += 1
 
         # full pathname of tarball
         tb = Path(tbmd5[0:-4])
         tbmd5 = Path(tbmd5)
-
-        # directory
         tbdir = tb.parent
-
-        # resultname: get the basename foo.tar.xz and then strip the .tar.xz
-        resultname = tb.name
-
         controller = tbdir.name
-        dest = archive / controller
-
-        # Create a new dataset tracker in UPLOADING state, and add it to the
-        # database.
-        #
-        # NOTE: Technically, this particular workflow has no "UPLOADING" as
-        # the `pbench-server-prep-shim-002` command isn't invoked until the
-        # tarball and MD5 has been entirely uploaded by the agent via `ssh`;
-        # this method however can't be supported once we have authorized user
-        # ownership, and the model fits the server `PUT` method where an
-        # unexpected termination could leave a tarball in "Uploading" state.
-        #
-        # TODO: We have no way to identify an owner here, so assign it to
-        # the arbitrary "pbench" user. This will go away when we drop this
-        # component entirely in favor of PUT.
```

Comment:
Happy to see this code go. Once this is in, I can change the Dataset constructor to require an MD5.